src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -150,23 +139,25 @@ public static DataGroup convertFitsToDataGroup(String fits_filename,
 
             if (table_idx==0) { //FITS tables are not at 0, if at zero try to read it as an image first
                 dg= getFitsImageAsTable(hdu,metaInfo);
-                if (dg==null && hdus.length>1) {
+                result = getFitsImageAsTable(hdu,metaInfo);
+                if (result==null && hdus.length>1) {
                     FitsFactory.setLongStringsEnabled(false);
-                    StarTable table = getStarTable(hdus[1], fits_filename, table_idx); // this is really a hack, maybe it should not be here.
-                    dg= convertStarTableToDataGroup(table, dataCols, headerCols, strategy);
+                    //StarTable table = getStarTable(hdus[1], fits_filename, table_idx); // this is really a hack, maybe it should not be here.
+                    result = readFitsTable(hdu, fits_filename, dataCols, headerCols, strategy);
+                    //dg= convertStarTableToDataGroup(table, dataCols, headerCols, strategy);
                 }
             }
             else { // if >0 then try to read it as a FITS table first.
                 FitsFactory.setLongStringsEnabled(false);
                 StarTable table = getStarTable(hdu, fits_filename, table_idx);
+                result = readFitsTable(hdu, fits_filename, dataCols, headerCols, strategy);
```

Comment:
Forgot to comment out `StarTable table = getStarTable(hdu, fits_filename, table_idx);`?

Reply:
Yes I've kept the StarTable stuff only for the draft PR..but I'll remove the code (including the `getStarTable` and `converStarTableToDataGroup` functions and their dependent functions as well) when I turn this into a PR (probably in next push to the branch)